sql/vector_mhnsw.cc

```diff
@@ -188,7 +188,56 @@ struct FVector
   }
 #endif
 
+
+/*
+  ARM NEON implementation. A microbenchmark shows 1.7x dot_product() performance
+  improvement compared to regular -O2/-O3 builds and 2.4x compared to builds
+  with auto-vectorization disabled.
+
+  There seem to be no performance difference between vmull+vmull_high and
+  vmull+vmlal2_high implementations. The second implementation is preserved
+  just for reference.
```

Comment:
let's drop the other implementation :) just keep the comment "There seem to be ... implementation"

Reply:
Alright, I'll let the other implementation gather dust as jira comment.